src/worker.py

```
@@ -1296,6 +1321,44 @@ async def _dispatch(request, env):
+    enr = await env.DB.prepare(
+        "SELECT id FROM enrollments WHERE activity_id = ? AND user_id = ? AND status = 'active'"
+    ).bind(sess["activity_id"], user["id"]).first()
+    if not enr:
+        return err("You must be enrolled in this activity", 403)
+    # Upsert attendance
+    try:
+        await env.DB.prepare(
+            "INSERT INTO session_attendance (id, session_id, user_id, status) VALUES (?, ?, ?, ?)"
+            " ON CONFLICT(session_id, user_id) DO UPDATE SET status = excluded.status"
+        ).bind(new_id(), session_id, user["id"], status).run()
+    except Exception as exc:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)